File: services/API/Domain/Address.cs
using System;

namespace Domain
{
    public class Address
    {
        public Guid Id { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string Phone { get; set; }
        public string ApartmentNumber { get; set; }
        public string StreetAddress { get; set; }
        public int DistrictId { get; set; }
        public int ProvinceId { get; set; }
        public string WardName { get; set; }
        public string WardCode { get; set; }
        public string DistrictName { get; set; }
        public string ProvinceName { get; set; }
        public bool IsMain { get; set; }
    }
}

File: services/web/src/model/address.ts
export type AddressFormSchema = {
  id?: string;
  firstName: string;
  lastName: string;
  phoneNumber: string;
  apartmentNumber: string;
  province: {
    id: number;
    name: string;
    code: string;
  };
  district: {
    id: number;
    name: string;
    code: string;
  };
  ward: {
    id: number;
    name: string;
    code: string;
  };
  street: string;
  isDefault: boolean;
};

export type Address = {
  id: string;
  firstName: string;
  lastName: string;
  phone: string;
  apartmentNumber: string;
  streetAddress: string;
  districtId: number;
  provinceId: number;
  wardName: string;
  wardCode: string;
  districtName: string;
  provinceName: string;
  isMain: boolean;
};

export type InputValidation = {
  value: any;
  onBlur: boolean;
  ruleNames: string[];
};
export type AddressFormValidationSchema = {
  id?: string;
  firstName: InputValidation;
  lastName: InputValidation;
  phoneNumber: InputValidation;
  apartmentNumber: InputValidation;
  province: {
    id: number;
    name: string;
    code: string;
  };
  district: {
    id: number;
    name: string;
    code: string;
  };
  ward: {
    id: number;
    name: string;
    code: string;
  };
  street: string;
  isDefault: boolean;
};
